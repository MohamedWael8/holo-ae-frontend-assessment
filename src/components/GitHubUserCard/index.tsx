import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  StyledGitHubCard,
  GitHubCardMeta,
  GitHubOwnerImage,
  GitHubUserGeneralContainer,
  GitHubUserBasicInfo,
} from "./styles";

type GitHubUserCardProp = {
  name: string;
  url: string;
  ownerAvatar: string;
  bio: string;
  location: string;
  repos: number;
  followers: number;
};

const GitHubUserCard: React.FC<GitHubUserCardProp> = ({
  name,
  url,
  ownerAvatar,
  bio,
  location,
  repos,
  followers,
}) => (
  <StyledGitHubCard href={url} data-github={name} target="_blank">
    <GitHubUserGeneralContainer>
      <GitHubOwnerImage src={ownerAvatar} />
      <GitHubUserBasicInfo>
        <h3>{name}</h3>
        <p>{bio}</p>
      </GitHubUserBasicInfo>
    </GitHubUserGeneralContainer>
    <GitHubCardMeta>
      <FontAwesomeIcon icon="location" />
      {location}
    </GitHubCardMeta>
    <GitHubCardMeta>
      <FontAwesomeIcon icon="book-bookmark" />
      <span>{repos}</span>
    </GitHubCardMeta>
    <GitHubCardMeta>
      <FontAwesomeIcon icon="user-group" />
      <span>{followers}</span>
    </GitHubCardMeta>
  </StyledGitHubCard>
);

export default GitHubUserCard;
