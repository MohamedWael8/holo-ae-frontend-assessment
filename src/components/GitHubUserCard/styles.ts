import styled from "styled-components";

export const StyledGitHubCard = styled.a`
  box-sizing: border-box;
  border: 1px solid #ccc;
  margin: 10px;
  padding: 20px;
  color: #555;
  text-decoration: none;
  font-size: 13px;
  width: 90%;
  min-width: 250px;
  justify-content: space-between;
`;

export const GitHubUserGeneralContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin-bottom: 10px;
`;

export const GitHubUserBasicInfo = styled.div`
  display: flex;
  flex-direction: column;

  > h3 {
    margin-top: 0;
    padding-top: 10px;
    margin-bottom: 5px;
    color: #4078c0;
    font-size: 15px;
  }

  > p {
    margin-top: 2px;
    margin-bottom: 2px;
  }
`;

export const GitHubOwnerImage = styled.img`
  border-radius: 50%;
  width: 75px;
  height: 75px;
  margin-right: 1rem;
`;

export const GitHubCardMeta = styled.span`
  margin-right: 20px;

  > svg {
    margin-right: 4px;
  }
`;
