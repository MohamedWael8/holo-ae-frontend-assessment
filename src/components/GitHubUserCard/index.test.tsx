import renderer from "react-test-renderer";
import GitHubUserCard from "./index";

it("renders correctly", () => {
  const testUser = {
    avatar_url: "https://avatars.githubusercontent.com/u/1836624?v=4",
    events_url: "https://api.github.com/users/hello/events{/privacy}",
    followers_url: "https://api.github.com/users/hello/followers",
    following_url: "https://api.github.com/users/hello/following{/other_user}",
    gists_url: "https://api.github.com/users/hello/gists{/gist_id}",
    gravatar_id: "",
    html_url: "https://github.com/hello",
    id: 1836624,
    login: "hello",
    node_id: "MDEyOk9yZ2FuaXphdGlvbjE4MzY2MjQ=",
    organizations_url: "https://api.github.com/users/hello/orgs",
    received_events_url: "https://api.github.com/users/hello/received_events",
    repos_url: "https://api.github.com/users/hello/repos",
    score: 1,
    site_admin: false,
    starred_url: "https://api.github.com/users/hello/starred{/owner}{/repo}",
    subscriptions_url: "https://api.github.com/users/hello/subscriptions",
    type: "Organization",
    url: "https://api.github.com/users/hello",
    bio: "test bio",
    repository_count: 33,
    followers: 2,
    location: "Chicago, Il",
  };
  const tree = renderer
    .create(
      <GitHubUserCard
        key={testUser.login}
        name={testUser.login}
        url={testUser.html_url}
        ownerAvatar={testUser.avatar_url}
        bio={testUser.bio}
        location={testUser.location}
        repos={testUser.repository_count}
        followers={testUser.followers}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
