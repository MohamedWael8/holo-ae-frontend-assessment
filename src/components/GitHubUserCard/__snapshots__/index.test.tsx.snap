// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="sc-erFrCC PToJc"
  data-github="hello"
  href="https://github.com/hello"
  target="_blank"
>
  <div
    className="sc-erzdQh ifAKBa"
  >
    <img
      className="sc-jknFwr bSCqBc"
      src="https://avatars.githubusercontent.com/u/1836624?v=4"
    />
    <div
      className="sc-eERUzQ eqHLFK"
    >
      <h3>
        hello
      </h3>
      <p>
        test bio
      </p>
    </div>
  </div>
  <span
    className="sc-jxGwga fAKfrK"
  >
    Chicago, Il
  </span>
  <span
    className="sc-jxGwga fAKfrK"
  >
    <span>
      33
    </span>
  </span>
  <span
    className="sc-jxGwga fAKfrK"
  >
    <span>
      2
    </span>
  </span>
</a>
`;
